### PR TITLE
chore: Move trackedEntityAttributeValue service and store into tracker [DHIS2-17714]

### PR DESCRIPTION
**Pull request by enricocolasante**
## Big picture

Tracker has multiple services for each entity like tracked entity, enrollment, event and relationship. One is in dhis-api and one in dhis-service-tracker. Our goal is to provide one API (service) per entity that is part of the tracker domain/team.

Trackers architecture splits read/write into exporter services and an importer. So if you want to get data you use an exporter. If you want to insert, update or delete you have to go through the importer. Only then can we ensure integrity of tracker data.

Another goal is that code that provides tracker functionality and is thus owned by the tracker team lives in ideally one maven module (We can settle on a name later on maybe `dhis-service-tracker` or `dhis-tracker`).

This is a **big** task that we are going to implement in many small steps.

## This PR

- Move `TrackedEntityAttributeValueService`, `DefaultTrackedEntityAttributeValueService` and `HibernateTrackedEntityAttributeValueStore` into tracker.
- Remove `TrackedEntityAttributeValueStore` interface as the store is only used by the service.
- Change `TrackedEntityAttributeValueDeletionHandler` to directly call the DB
- Remove unused methods

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentityattributevalue/TrackedEntityAttributeValueDeletionHandler.java [hunk 1]
```diff
@@ -40,21 +39,19 @@
...
   protected void register() {
     whenVetoing(TrackedEntityAttribute.class, this::allowDeleteTrackedEntityAttribute);
   }
 
   private DeletionVeto allowDeleteTrackedEntityAttribute(TrackedEntityAttribute attribute) {
-    return attributeValueService.getCountOfAssignedTrackedEntityAttributeValues(attribute) == 0
-        ? ACCEPT
-        : VETO;
+    String sql =
+        "select 1 from trackedentityattributevalue where trackedentityattributeid = :id limit 1";
```
Comment (muilpp) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentityattributevalue/TrackedEntityAttributeValueDeletionHandler.java:54]: do we need to add "limit 1" if `trackedentityattributeid` is the PK?

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/trackedentityattributevalue/HibernateTrackedEntityAttributeValueStore.java [hunk 2]
```diff
@@ -38,15 +37,14 @@
...
-import org.hisp.dhis.trackedentityattributevalue.TrackedEntityAttributeValueStore;
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
 
 /**
  * @author Abyot Asalefew
  */
-@Repository("org.hisp.dhis.trackedentityattributevalue.TrackedEntityAttributeValueStore")
+@Repository
```
Comment (muilpp) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/trackedentityattributevalue/HibernateTrackedEntityAttributeValueStore.java:47]: even though most of the time we have only one implementation of a service and a store, we usually give it a name, I took it almost as a convention in this project... so this is just to understand why in this case the class name is enough

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentityattributevalue/TrackedEntityAttributeValueDeletionHandler.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentityattributevalue/TrackedEntityAttributeValueDeletionHandler.java:54]: Yes, this is a one-to-many relationship.
For a `TrackedEntityAttribute` we can have multiple `TrackedEntityAttributeValue`s, and we just want to check that it exists

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/trackedentityattributevalue/HibernateTrackedEntityAttributeValueStore.java (hunk 2 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/trackedentityattributevalue/HibernateTrackedEntityAttributeValueStore.java]: No reason, I was following what I did for the service.
Putting it back

**Review by muilpp — APPROVED** [review]

**Review by ameenhere — APPROVED** [review]

**Pull request merged**